gcc/common/config/riscv/riscv-common.c

```diff
@@ -33,6 +33,61 @@ along with GCC; see the file COPYING3.  If not see
 
 #define RISCV_DONT_CARE_VERSION -1
 
+/* ISA extension name class. E.g. "zbb" corresponds to RV_ISA_CLASS_Z,
+   "xargs" corresponds to RV_ISA_CLASS_X, etc.  */
+
+typedef enum riscv_isa_ext_class
+  {
+   RV_ISA_CLASS_UNKNOWN,
+   RV_ISA_CLASS_X,
+   RV_ISA_CLASS_S,
+   RV_ISA_CLASS_SX,
+   RV_ISA_CLASS_Z
+  } riscv_isa_ext_class_t;
+
+/* Classify the argument 'arch' into one of riscv_isa_ext_class_t.  */
+
+static riscv_isa_ext_class_t
+riscv_get_prefix_class (const char *arch)
+{
+  switch (*arch)
+    {
+    case 's':
+      if (strlen (arch) > 1 && arch[1] == 'x')
+	return RV_ISA_CLASS_SX;
+      else
+	return RV_ISA_CLASS_S;
+
+    case 'x': return RV_ISA_CLASS_X;
+    case 'z': return RV_ISA_CLASS_Z;
+    default: return RV_ISA_CLASS_UNKNOWN;
+    }
+}
+
+/* Structure describing parameters to use when parsing a particular
+   riscv_isa_ext_class_t. One of these should be provided for each
+   possible class, except RV_ISA_CLASS_UNKNOWN.  */
+
+typedef struct riscv_parse_config
+{
+  /* Class of the extension. */
+  riscv_isa_ext_class_t ext_class;
+
+  /* Lower-case prefix string for error printing
+     and internal parser usage, e.g. "sx", "z".  */
+  const char *prefix_lower;
+
+  /* Upper-case prefix string for error printing,
+     e.g. "SX". Used as "Error: SX extension ..."  */
+  const char *prefix_upper;
```

Comment:
Like review comment in binutils side, just merge `prefix_lower` and `prefix_upper` is fine.